hydroflow/examples/vector_clock/server.rs

```diff
@@ -21,22 +21,21 @@ pub(crate) async fn run_server(outbound: UdpSink, inbound: UdpStream, opts: crat
 
         // merge in the msg vc to the local vc
         inbound_chan[merge] -> map(|(msg, _addr): (EchoMsg, SocketAddr)| msg.vc) -> mergevc;
-        mergevc = fold::<'static> (bot_vc,
-                    |mut old: <VecClock as LatticeRepr>::Repr, vc: <VecClock as LatticeRepr>::Repr| {
-                            let my_addr = format!("{:?}", opts.addr.unwrap());
-                            let bump = HashMap::from([(my_addr.clone(), old[&my_addr] + 1)]);
-                            <VecClock as Merge<VecClock>>::merge(&mut old, bump);
-                            <VecClock as Merge<VecClock>>::merge(&mut old, vc);
-                            old.clone()
-                    }
-                );
+        mergevc = fold::<'static> (VecClock::default(), |mut old, vc| {
```

Comment:
Semantics of fold init value are confusing, how did this work before? implicit clone?
"pass by source"